File: Leetcode-Python-solution/Two-Sum/TwoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = {}
        for i, num in enumerate(nums):
            x = target - num
            if x in m:
                return [m[x], i]
            else:
                m[num] = i
        return [-1, -1]

File: Leetcode-CPP-solution/Two-Sum/TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        map <int,int> m;
        
        for(int i = 0;i < n;i++) {
            int x = target - nums[i];
            
            if(m.find(x) != m.end()) {
                return {i,m[x]};
            } else {
                m[nums[i]] = i;
            }
            
        }
        return {-1,-1};
    }
};